src/main/java/com/amazon/opendistroforelasticsearch/sql/query/DefaultQueryAction.java

```diff
@@ -223,15 +235,45 @@ private void setSorts(List<Order> orderBys) {
 
                     request.addSort(
                             SortBuilders
-                                    .scriptSort(new Script(order.getName()), getScriptSortType(order))
-                                    .order(SortOrder.valueOf(order.getType())));
+                                    .scriptSort(new Script(orderByName), getScriptSortType(order))
+                                    .order(sortOrder));
                 } else {
-                    request.addSort(order.getName(), SortOrder.valueOf(order.getType()));
+                    if (orderByName.equals(ScoreSortBuilder.NAME)) {
+                        request.addSort(orderByName, sortOrder);
+                    } else {
+                        FieldSortBuilder fieldSortBuilder;
+                        if (sortBuilderMap.containsKey(orderByName)) {
+                            fieldSortBuilder = sortBuilderMap.get(orderByName);
+                            setSortParmas(fieldSortBuilder, order);
+                        } else {
+                            fieldSortBuilder = SortBuilders.fieldSort(orderByName);
+                            setSortParmas(fieldSortBuilder, order);
+                            sortBuilderMap.put(orderByName, fieldSortBuilder);
+                            request.addSort(fieldSortBuilder);
```

Comment:
Assuming that there are no nasty side effects, feels like that computeIfAbsent could make code a bit concise and simpler

```
                        FieldSortBuilder fieldSortBuilder = sortBuilderMap.computeIfAbsent(orderByName, key -> {
                            FieldSortBuilder fs = SortBuilders.fieldSort(key);
                            request.addSort(fs);
                            return fs;
                        });

                        setSortParmas(fieldSortBuilder, order);
```

Reply:
Changed as suggested.